cognite_toolkit/cdf.py

```diff
@@ -694,130 +695,31 @@ def main_init(
-        exit(1)
+    project_dir: ProjectDirectory
+    if upgrade:
+        project_dir = ProjectDirectoryUpgrade(Path.cwd() / f"{init_dir}", dry_run)
+    else:
+        project_dir = ProjectDirectoryInit(Path.cwd() / f"{init_dir}", dry_run)
+    verbose = ctx.obj.verbose
 
-    if not dry_run and not upgrade:
-        project_dir.mkdir(exist_ok=True)
+    project_dir.set_source(git_branch)
 
-    if upgrade:
-        print("  Will upgrade modules and files in place.")
-
-    copy_prefix = "Would" if dry_run else "Will"
-    print(f"{copy_prefix} copy these files to {target_dir_display}:")
-    print(files_to_copy)
-    modules_by_root: dict[str, list[str]] = {}
-    for root_module in root_modules:
-        modules_by_root[root_module] = [
-            f"{module.relative_to(template_source)!s}" for module, _ in iterate_modules(template_source / root_module)
-        ]
-
-        print(f"{copy_prefix} copy these modules to {target_dir_display} from {root_module}:")
-        print(modules_by_root[root_module])
-
-    copy_prefix = "Would copy" if dry_run else "Copying"
-    for filename in files_to_copy:
-        if ctx.obj.verbose:
-            print(f"{copy_prefix} file {filename} to {target_dir_display}")
-        if not dry_run:
-            if filename == "README.md":
-                content = (template_source / filename).read_text().replace("<MY_PROJECT>", init_dir)
-                (project_dir / filename).write_text(content)
-            else:
-                shutil.copyfile(template_source / filename, project_dir / filename)
-
-    if upgrade and not no_backup:
-        prefix = "Would have backed up" if dry_run else "Backing up"
-        if ctx.obj.verbose:
-            print(f"{prefix} {target_dir_display}")
-        if not dry_run:
-            backup_dir = tempfile.mkdtemp(prefix=f"{project_dir.name}.", suffix=".bck", dir=Path.cwd())
-            shutil.copytree(project_dir, Path(backup_dir), dirs_exist_ok=True)
-    elif upgrade:
-        print(
-            f"[bold yellow]WARNING:[/] --no-backup is specified, no backup {'would have been' if dry_run else 'will be'} be."
-        )
+    project_dir.create_project_directory(clean)
 
-    for root_module in root_modules:
-        if ctx.obj.verbose:
-            print(f"{copy_prefix} the following modules from  {root_module} to {target_dir_display}")
-            print(modules_by_root[root_module])
-        if not dry_run:
-            (Path(project_dir) / root_module).mkdir(exist_ok=True)
-            # Default files are not copied, as they are only used to setup the config.yaml.
-            shutil.copytree(
-                template_source / root_module,
-                project_dir / root_module,
-                dirs_exist_ok=True,
-                ignore=shutil.ignore_patterns("default.*"),
-            )
+    if isinstance(project_dir, ProjectDirectoryUpgrade):
+        project_dir.do_backup(no_backup, verbose)
```

Comment:
Do we need backup now or can we get rid of it? Everything we can remove is great!

Reply:
We still do a change of cognite modules if it has not been touched, which I guess was the reason to have the backup in the first place, thus the reason for the backup has not changed.